cardinal/world.go

```diff
@@ -173,115 +214,89 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 	return world, nil
 }
 
-// CreateMany creates multiple entities in the world, and returns the slice of ids for the newly created
-// entities. At least 1 component must be provided.
-func CreateMany(wCtx WorldContext, num int, components ...component.Component) ([]EntityID, error) {
-	ids, err := ecs.CreateMany(wCtx.Engine(), num, components...)
-	if wCtx.Engine().IsReadOnly() || err == nil {
-		return ids, err
-	}
-	return nil, logAndPanic(wCtx, err)
-}
-
-// Create creates a single entity in the world, and returns the id of the newly created entity.
-// At least 1 component must be provided.
-func Create(wCtx WorldContext, components ...component.Component) (EntityID, error) {
-	id, err := ecs.Create(wCtx.Engine(), components...)
-	if wCtx.Engine().IsReadOnly() || err == nil {
-		return id, err
-	}
-	return 0, logAndPanic(wCtx, err)
+func (w *World) CurrentTick() uint64 {
+	return w.tick.Load()
 }
 
-// SetComponent Set sets component data to the entity.
-func SetComponent[T component.Component](wCtx WorldContext, id entity.ID, comp *T) error {
-	err := ecs.SetComponent[T](wCtx.Engine(), id, comp)
-	if wCtx.Engine().IsReadOnly() || err == nil {
-		return err
-	}
-	if eris.Is(err, ErrEntityDoesNotExist) ||
-		eris.Is(err, ErrComponentNotOnEntity) {
-		return err
-	}
-	return logAndPanic(wCtx, err)
-}
+// Tick performs one game tick. This consists of taking a snapshot of all pending transactions, then calling
+// each System in turn with the snapshot of transactions.
+func (w *World) Tick(ctx context.Context) error {
+	// Check the current world state and handle accordingly
+	switch w.WorldState {
+	case WorldStateInit:
+		// If the engine is not ready, we don't want to tick the engine.
+		// Instead, make we have recovered the state of the world firstreturn eris.New("invalid world state to tick")
+	case WorldStateReady:
+		// If the world have loaded game state, we can set the it WorldStateRunning
+		w.WorldState = WorldStateRunning
+	case WorldStateRecovering:
+		// If the world is recovering, proceed
+	case WorldStateRunning:
+		// If the world is already running, proceed
+	}
```

Comment:
nit: cases that do the same thing can be grouped:

```suggestion
	case WorldStateRecovering, WorldStateRunning:
		// If the world is running or recovering, proceed
	}
```